plugins/colorbutton/plugin.js

```diff
@@ -503,22 +516,29 @@
 			 * @returns {String} Returns color in hex format, but without the hash at the beginning, e.g. `ff0000` for red.
 			 */
 			normalizeColor: function( color ) {
-				var alphaRegex = /^(rgb|hsl)a\(/g,
-					transparentRegex = /^rgba\((\s*0\s*,?){4}\)$/g,
-					isAlphaColor = alphaRegex.test( color ),
-					// Browsers tend to represent transparent color as rgba(0, 0, 0, 0), so we need to filter out this value.
-					isTransparentColor = transparentRegex.test( color ),
-					colorInstance;
-
-				// For colors with alpha channel we need to use CKEDITOR.tools.color normalization (#4351).
-				if ( isAlphaColor && !isTransparentColor ) {
-					colorInstance = new CKEDITOR.tools.color( color );
-
-					return CKEDITOR.tools.normalizeHex( colorInstance.getHex() || '' ).replace( /#/g, '' );
+				var colorInstance = ( color instanceof CKEDITOR.tools.color ) ? color : new CKEDITOR.tools.color( color, '' ),
+					isTransparent = Number( colorInstance._.alpha ) === 0;
+
+				if ( isTransparent ) {
+					return 'transparent';
```

Comment:
Any somehow transparent color should be converted to 'transparent'? Because API comments say this method returns hex without '#'. The previous version seems to skip this case at all?

Reply:
Yes, the previous version returns incorrect hex value for transparent colors. This version returns `transparent` explicitly.